### PR TITLE
Add Package Manager Security Landscape Survey

### PR DESCRIPTION
**Pull request by lumjjb**
After review of the group, as an action item from the meeting from 30 Nov 2022, here is the PR to add the work done for the landscape survey to the repository!

### CODE REVIEW

File: survey/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# The Package Manager Security Landscape Survey
+
+In 2022, the OpenSSF Securing Software Repositories Working Group embarked on
+an effort to survey the current security landscape of package manager
+ecosystems.
+
+This effort provides a survey/landscape of different security mechanisms and
+features that are implemented across the different ecosystems as it pertains to
```
Comment (jchestershopify) [inline survey/README.md:8]: Nit: "as it pertains" -> "as they pertain"; the key word being "features", which is plural.

File: survey/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+overall security priorities, and to cross-pollinate ideas, threat models and
+designs across the ecosystems. 
+
+The survey covers 3 main broad topics: Secure, Ingest and
+Evaluate. This includes areas of:
+- Integrity
+- Typos & dependency confusion
+- Handling compromise and malice
+- Portecting the ecosystem
+- Auth and credential Management
```
Comment (jchestershopify) [inline survey/README.md:19]: Nit: should "credential" be "Credential"?

File: survey/summary.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+doesn't allow execution of arbitrary code on the host in an unprotected way.
+This is an area that requires further investigation into why that is not a
+priority and look into a way to be able to have isolation and provenance of
+builds without this property.
+
+![image](https://user-images.githubusercontent.com/3060102/220681554-3b469808-3fdd-441e-b91c-0cf060461af2.png)
+
+## Protecting the ecosystem
+
+Most ecosystems are not currently performing any security testing against its
```
Comment (jchestershopify) [inline survey/summary.md:64]: Nit: "its" should be "their", as "ecosystems" is plural.

File: survey/summary.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+code/infrastructure. However, all of them are interested (7) or have it planned
+(2). These ecosystems could be funded for security testing.
+
+All ecosystems have a process in place for vulnerability disclosure or are
+interested in doing so.
+
+Only 1 ecosystem has additional supply chain security guarantees for the
+binaries/tooling that it produces. This is an area for improvement overall.
```
Comment (jchestershopify) [inline survey/summary.md:72]: This is a little vague, can it be expanded? Or would that reveal the ecosystem?

File: survey/summary.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+Generally there is interest in protecting token security, which is positive
+feedback. There are 2 ecosystems that currently have no plans on requiring MFA.
+Further investigation on why would be helpful.
+
+## Policy:
+
+With policy, there is a mix of implementations that help provide ability to
+create policy on keys/hashes as well as checking for known vulnerabilities, as
+well as a significant number of ecosystems (4-5) which do not but are
+interested.
```
Comment (jchestershopify) [inline survey/summary.md:87]: I think splitting this into 2, perhaps 3, sentences would make it easier to read.

File: survey/summary.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+
+## Identify / produce good dependencies
+
+In general most ecosystems allow pinned dependencies that are immutably
+verifiable. However, there are 3 ecosystems that do not have this yet. Further
+investigation on why would be helpful.
+
+Only 27.3% of ecosystems have fuzzing services and utilities available, 45.5%
+have no plans on implementing and the last 27.3% are interested in exploring.
```
Comment (jchestershopify) [inline survey/summary.md:116]: These don't sum to 100%. Which is to be expected given rounding error. Perhaps at the introduction we can warn that amounts will not sum to exactly 100%?

File: survey/summary.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+This could be an avenue for education of the value of fuzzing within the
+different ecosystems.
+
+There is a surprising lack of SAST/DAST services available for ecosystems to
+detect bad code patterns with 5 ecosystems without the capabilities at the
+moment. Funding of tooling could be helpful here.
```
Comment (jchestershopify) [inline survey/summary.md:122]: I think this could be split into 2 sentences.

File: survey/summary.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+There is a high demand (8) of services and tools for developers to help keep
```
Comment (jchestershopify) [inline survey/summary.md:124]: Nit: "high demand (8) of" should be "high demand (8) for".

**Review by jchestershopify — COMMENTED** [review]
Thank you for all the work you've done on this! 

I have some suggested changes. Also, could we put this under `survey/2022/` to give ourselves some future-proofing against future surveys taking place?

File: survey/summary.md (hunk 6 above)
Comment (lumjjb) [reply survey/summary.md:116]: image


:| lol... it's one of those rounding error things... i'll reduce it to 45.4 for consistency.

**Comment by lumjjb** [comment]
Thanks for the review @jchestershopify ! Addressed the comments!

File: survey/README.md (hunk 2 above)
Comment (lumjjb) [reply survey/README.md]: made everything lower case for consistency

File: survey/2022/README.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+security critical user journeys. The goal of this exercise is to understand
+overall security priorities, and to cross-pollinate ideas, threat models and
+designs across the ecosystems. 
+
+The survey covers 3 main broad topics: Secure, Ingest and
+Evaluate. This includes areas of:
+- Integrity
+- Typos & dependency confusion
+- Handling compromise and malice
+- Portecting the ecosystem
```
Comment (joshuagl) [inline survey/2022/README.md:18]:
```suggestion
- Protecting the ecosystem
```

File: survey/2022/README.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+- Auth and credential Management
```
Comment (joshuagl) [inline survey/2022/README.md:19]:
```suggestion
- Authentication and credential management
```

File: survey/2022/README.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+- Policy
+- SBOMs
+- Identifying and producing good depdencies
```
Comment (joshuagl) [inline survey/2022/README.md:22]:
```suggestion
- Identifying and producing good dependencies
```

File: survey/2022/README.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+ecosystems.
+
+This effort provides a survey/landscape of different security mechanisms and
+features that are implemented across the different ecosystems as it pertains to
+security critical user journeys. The goal of this exercise is to understand
+overall security priorities, and to cross-pollinate ideas, threat models and
+designs across the ecosystems. 
+
+The survey covers 3 main broad topics: Secure, Ingest and
+Evaluate. This includes areas of:
```
Comment (joshuagl) [inline survey/2022/README.md:14]: Would it make sense to group areas by topic?

File: survey/2022/summary.md [hunk 13]
```diff
@@ -0,0 +1,129 @@
+# Survey Summary and Takeaways
+
+This document summarizes the survey results and gives some insight into some
+takeaways from each category of the survey.
+
+In total, we had package manager maintainers (and/or reputable sources) from 11
+different package manager ecosystems responded to the survey. Including:
+
+-   PyPI (Python
```
Comment (joshuagl) [inline survey/2022/summary.md:9]:
```suggestion
-   PyPI (Python)
```

File: survey/2022/summary.md [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+-   Gradle (JVM)
+-   Basalt  (Bash)
```
Comment (joshuagl) [inline survey/2022/summary.md:11]:
```suggestion
-   Basalt (Bash)
```

File: survey/2022/summary.md [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
+-   Opam (OCaml)
+-   RubyGems (ruby)
```
Comment (joshuagl) [inline survey/2022/summary.md:13]:
```suggestion
-   RubyGems (Ruby)
```

File: survey/2022/summary.md [hunk 16]
```diff
@@ -0,0 +1,129 @@
...
+-   Npm (javascript)
```
Comment (joshuagl) [inline survey/2022/summary.md:14]:
```suggestion
-   npm (JavaScript)
```

File: survey/2022/summary.md [hunk 17]
```diff
@@ -0,0 +1,129 @@
...
+-   Pkg.go.dev (go)
```
Comment (joshuagl) [inline survey/2022/summary.md:15]:
```suggestion
-   Pkg.go.dev (Go)
```

File: survey/2022/summary.md [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+-   Cargo (Rust)
+-   Pub (Dart)
+-   Apache Maven
+-   Maven Central
```
Comment (joshuagl) [inline survey/2022/summary.md:19]: Are these both for Java/JVM packages?

File: survey/2022/summary.md [hunk 19]
```diff
@@ -0,0 +1,129 @@
...
+
+## Integrity
+
+Most support is still around developer owned keys for signing (with PGP being
+the most supported signature type), with some a couple ecosystems with no
+support for signing yet - but there is interest/plans.
```
Comment (joshuagl) [inline survey/2022/summary.md:25]:
```suggestion
the most supported signature type), with a couple ecosystems which do not
support signing yet - but there is interest/plans.
```

File: survey/2022/summary.md [hunk 20]
```diff
@@ -0,0 +1,129 @@
...
+
+![image](https://user-images.githubusercontent.com/3060102/220680273-500cc331-728b-48e6-a8e6-8c5fd131b57a.png)
+
+## Handling compromise and malice
+
+Most ecosystems don't see the responsibility of finding malware in the packages
+as their responsibility. However, they provide the ability to flag and handle
+malware in releases when it happens.
+
+Rust, Ruby and Gradle have no plans on ensuring that install and build code
```
Comment (joshuagl) [inline survey/2022/summary.md:54]:
```suggestion
Rust, Ruby and Gradle have no plans on ensuring that install and build of code
```

File: survey/2022/summary.md [hunk 21]
```diff
@@ -0,0 +1,129 @@
...
+
+All ecosystems have a process in place for vulnerability disclosure or are
+interested in doing so.
+
+Only 1 ecosystem has additional supply chain security guarantees for the
+binaries/tooling that it produces. This is an area for improvement overall.
+
+![image](https://user-images.githubusercontent.com/3060102/220682148-2d488135-4695-41b5-a6ab-e5f85883f612.png)
+
+## Auth and Credential Management
```
Comment (joshuagl) [inline survey/2022/summary.md:76]:
```suggestion
## Authentication and credential management
```

File: survey/2022/summary.md [hunk 22]
```diff
@@ -0,0 +1,129 @@
...
+
+![image](https://user-images.githubusercontent.com/3060102/220683454-3d685cb4-861a-498c-b3b9-6fbde42b7d3a.png)
+
+## SBOMs
+
+7 of 11 ecosystems do not currently have support for generating SBOM but are
+interested (5) or planned (2). Additional tooling support is required and could
+be a place where the OpenSSF SBOMs everywhere workstream 9 can be helpful.
+
+Most SBOM formats are around SPDX and CycloneDX, with the exception of golang
```
Comment (joshuagl) [inline survey/2022/summary.md:101]:
```suggestion
Most SBOM formats are around SPDX and CycloneDX, with the exception of Go
```

File: survey/2022/summary.md [hunk 23]
```diff
@@ -0,0 +1,129 @@
...
+
+![image](https://user-images.githubusercontent.com/3060102/220682148-2d488135-4695-41b5-a6ab-e5f85883f612.png)
+
+## Auth and Credential Management
+
+Generally there is interest in protecting token security, which is positive
+feedback. There are 2 ecosystems that currently have no plans on requiring MFA.
+Further investigation on why would be helpful.
+
+## Policy:
```
Comment (joshuagl) [inline survey/2022/summary.md:82]:
```suggestion
## Policy
```

File: survey/2022/summary.md [hunk 24]
```diff
@@ -0,0 +1,129 @@
...
+making it mandatory.
+
+![image](https://user-images.githubusercontent.com/3060102/220680128-180c076c-31de-4f32-884e-7c2de8137dc0.png)
+
+## Typos & dependency confusion
+
+45.5% of package managers do not require and do not plan on having DNS
+verification proof when reserving/exercising a domain/namespace or domain name
+attribute of a package. This could lead to impersonation of
+organisations/entities. A method to reduce the cost of protecting against this
```
Comment (joshuagl) [inline survey/2022/summary.md:39]:
```suggestion
organizations/entities. A method to reduce the cost of protecting against this
```
Nit: this is the en_GB spelling of organisations, but the rest of the doc is en_US.

**Review by joshuagl — COMMENTED** [review]
This is great, thanks so much for running the survey and summarising the results!

**Review by simi — APPROVED** [review]

File: survey/2022/README.md (hunk 12 above)
Comment (lumjjb) [reply survey/2022/README.md:14]: i think there's some overlap, so picking to just remove these 3 main topics.. they were used in generating the questions but i don't think they are particularly that helpful as part of the results

File: survey/2022/summary.md (hunk 18 above)
Comment (lumjjb) [reply survey/2022/summary.md:19]: yep - adding (Java) and then changing grade to (Java) as well for consistency

**Comment by lumjjb** [comment]
@joshuagl @simi thanks for the review! Addressed the comments!

**Comment by lumjjb** [comment]
Addressed all the comments. PTAL!

**Review by di — APPROVED** [review]

**Review by znewman01 — APPROVED** [review]

**Comment by di** [comment]
Thanks @lumjjb!

**Pull request merged**